<h1>We Rate Dogs: Report</h1>
<h4>By Luis Alarcon</h4>

<h2>Introduction</h2>

<p>The project belongs to the second term of the Udacity Data Analyst Nanodegree. This project starts with three main datasets:</p>

<ul>
    <li><b>Image Predictions</b>: Udacity provides this file.</li>
    <li><b>Twitter Archive</b>: Udacity provides this file.</li>
    <li><b>Twitter related to WeRateDogs</b>: This is obtained through a Twitter Development API.</li>
</ul>

<h3>Datasets: General Characteristics</h3>

<h4>Image Predictions</h4>
<p>The image predictions dataset has 2075 entries, with 12 columns with an assortment of types.</p>

<h4>Twitter Archive</h4>
<p>The Twitter Archive dataset has 2356 entries with 17 columns with an assortment of types.</p>

<h4>Twitter JSON</h4>
<p>The Twitter JSON dataset has 2341 entries with seven selected columns with an object and int64 types. Also, in order to gather the data for this dataset, we distinguished whether the twitter ID from the Twitter Archive dataset was found using the Twitter Development API; if not found then the twitter ID will be stored in a separate file.</p>

<h3>Quality Analysis</h3>

<h4>Null Values</h4>

</p>There are no null values in any of the datasets gathered.</p>

<h4>Data Problems</h4>

<p>After analysing the Twitter Archive dataset, some issues were found:</p>

<ul>
    <li><b>tweet_id:</b> The data type needs to be changed from int64 to object.</li>
    <li><b>in_reply_to_status_id:</b> erroneous data types.</li>
    <li><b>in_reply_to_user_id:</b> erroneous data types.</li>
    <li><b>name:</b> check if the name of the dog is the one mentioned in the tweet text. if it is not found remove it</li>
    <li><b>No retweets:</b> remove retweet data as well as columns associated with retweets.</li>
    <li><b>tweets with ratings with images.</b></li>
    <li><b>timestamp:</b> Change to datetime data type</li>
    <li><b>doggo, floofer, pupper, puppo:</b> change to one variable.</li>
    <li><b>rating:</b> Create a new column name rating which is the result of rating_numerator/rating_denomintor.</li>
</ul>

<p>All the errors found were amended.</p>

<p>Another dataset that presented issues after the initial analysis was the Image Prediction dataset. These were the issues found:</p>

<ul>
    <li><b>p1:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li>
    <li><b>p2:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li>
    <li><b>p3:</b> Some of the breeds predicted are not breeds. Therefore we will find whether the breeds predicted belong from a list made.</li> 
</ul>

<p>All issues found were amended/</p>

<p>Lastly, we fused the Twitter Archive and the Twitter JSON file, and we stored all modifications into new DataFrames and new files.</p>

<h3>Objectives</h3>

<p>The objective of this project is to understand the data provided to the students. The main focus of this report will be to analyse the primary dataset for values within the favourite count, retweet count and how that does interact with the rating provided, as well as the rating distribution.</p>

<p>Another aspect that we will only look at is the dog breed information in the image prediction file.</p>

<h2>Data Analysis</h2>

<p>The analysis started by finding the standard analysis of the Dataframe w_t_archive or working Twitter Archive file; the fusion between the Twitter Archive and Twitter JSON file.</p>


In [2]:
import pandas as pd

w_t_archive=pd.read_csv("./datasets/final_twitter-archive-enhanced.csv")
df_i_pre=pd.read_csv("./datasets/final_image-predictions.csv")

In [3]:
w_t_archive.describe()

tweet_id  in_reply_to_status_id  in_reply_to_user_id       rating  \
count  1.447000e+03           1.000000e+00         1.000000e+00  1447.000000   
mean   7.417931e+17           6.678065e+17         4.196984e+09     1.192581   
std    6.856868e+16                    NaN                  NaN     4.648617   
min    6.660293e+17           6.678065e+17         4.196984e+09     0.200000   
25%    6.782950e+17           6.678065e+17         4.196984e+09     1.000000   
50%    7.186133e+17           6.678065e+17         4.196984e+09     1.100000   
75%    7.961149e+17           6.678065e+17         4.196984e+09     1.200000   
max    8.924206e+17           6.678065e+17         4.196984e+09   177.600000   

       favorite_count  retweet_count  
count     1447.000000    1447.000000  
mean      9000.928127    2609.447132  
std      11881.523847    3947.478723  
min        105.000000      20.000000  
25%       2149.500000     638.500000  
50%       4494.000000    1433.000000  
75%      11534.500000    3145.500000  
max     123018.000000   60162.000000

<p>It shows that the mean rating is 1.93 and the mean favourite count is 9001.8, and the mean retweet count is 2609.90. Further information is shown in the table presented before.</p>

<p>The Image Prediction has 3 columns named:</p>

<ul>
    <li>p1</li>
    <li>p2</li>
    <li>p3</li>
</ul>

In [4]:
p1_count = pd.read_csv("./datasets/breed_p1_top10.csv")
p2_count = pd.read_csv("./datasets/breed_p2_top10.csv")
p3_count = pd.read_csv("./datasets/breed_p3_top10.csv")

In [8]:
print(p1_count)

                breed  count
0    golden_retriever    150
1  Labrador_retriever    100
2            Pembroke     89
3           Chihuahua     83
4                 pug     57
5                chow     44
6             Samoyed     43
7          toy_poodle     39
8          Pomeranian     38
9      cocker_spaniel     30


In [10]:
print(p2_count)

                      breed  count
0        Labrador_retriever    104
1          golden_retriever     92
2                 Chihuahua     44
3                Pomeranian     42
4  Chesapeake_Bay_retriever     41
5            French_bulldog     41
6                toy_poodle     37
7            cocker_spaniel     34
8            Siberian_husky     33
9          miniature_poodle     33


In [9]:
print(p3_count)

                       breed  count
0         Labrador_retriever     79
1                  Chihuahua     58
2           golden_retriever     48
3                 Eskimo_dog     38
4                     kuvasz     34
5  Staffordshire_bullterrier     32
6                       chow     32
7             cocker_spaniel     31
8                     beagle     31
9                 toy_poodle     29


<p>It seems to be that there is an overwhelming preference towards labrador retrievers and golden retrievers. The data from the image prediction dataset shows that those two breeds are the among the top 3 most common breeds. The third most common breed is the poodle, which in all three appears in the top three.</p>

<p>The first image set of data is shown in the following histogram. It shows that the rating with the highest value is 1.2, followed by 1.1 and 1.0. There are some ratings with a count of one, and these ratings are 2.7 177.6, 7.5 and 3.43.
</p>

<img src="./graphs/histogram_rating_count.png" />
<img src="./graphs/histogram_rating_count_ne.png" />
<img src="./graphs/histogram_rating_count_e.png" />

<p>The following information gathered is from the rating values vs the mean and retweet count. The scatter plot involves all the data and shows a high concentration towards value below 2; this is due to the majority of ratings been below 2.</p>

<img src="./graphs/scatter_plot_ratings_vs_mean_retweet_count.png" />

<p>A more refined image for ratings below 2 shows a gradual increase in retweet count in ratings higher than 1. There is a small increase in retweet count between ratings 0.4 and 0.6; however, there is an outlier between 0.6 to 0.7. As for ratings above 2, there is not a recognisable pattern.
</p>
<img src="./graphs/scatter_plot_ratings_vs_mean_retweet_count_ratings-bellow-2.png" />
<img src="./graphs/scatter_plot_ratings_vs_mean_retweet_count_ratings-above-2.png" />

<p>The last information provided here is from the rating values vs the mean favourite count. A scatter plot shows all the data and shows a high concentration towards value below 2; this is due to the majority of ratings been below 2.</p>

<img src="./graphs/scatter_plot_ratings_vs_mean_favorite_count.png" />

<p>The same pattern from the rating vs mean retweet count is shown in the data rating favourite count. </p>

<img src="./graphs/scatter_plot_ratings_vs_mean_favorite_count_bellow-2.png" />
<img src="./graphs/scatter_plot_ratings_vs_mean_favorite_count_above-2.png" />

<h2>Further development</h2>

<p>A lot more can be done with these data such as further analysis of location vs ratings as well as time vs ratings. Perhaps an increase in a rating is slightly influenced by the time or day</p>